frappe/core/doctype/user_permission/test_user_permission.py

```diff
@@ -185,6 +185,54 @@ def test_user_perm_for_nested_doctype(self):
 		self.assertTrue(has_user_permission(frappe.get_doc("Person", parent_record.name), user.name))
 		self.assertFalse(has_user_permission(frappe.get_doc("Person", child_record.name), user.name))
 
+	def test_user_perm_for_nested_doctype_case_2(self):
+		"""Test if user is shown doctype values if a User permission if applied to it."""
+		from frappe.core.doctype.doctype.test_doctype import new_doctype
+
+		user = create_user("nested_doc_user2@example.com")
+		if not frappe.db.exists("DocType", "Person"):
+			doc = new_doctype(
+				"Person",
+				fields=[{"label": "Person Name", "fieldname": "person_name", "fieldtype": "Data"}],
+				unique=0,
+			)
+			doc.is_tree = 1
+			doc.insert()
+
+		parent_record = frappe.get_doc(
+			{"doctype": "Person", "person_name": "Parent", "is_group": 1}
+		).insert()
+
+		child_record = frappe.get_doc(
+			{
+				"doctype": "Person",
+				"person_name": "Child",
+				"is_group": 0,
+				"parent_person": parent_record.name,
+			}
+		).insert()
+
+		add_user_permissions(get_params(user, "Person", parent_record.name))
+		frappe.set_user(user.name)
+
+		visible_names = frappe.get_list(
+			doctype="Person",
+			fields=["person_name"],
+		)
+
+		frappe.db.set_value(
+			"User Permission", {"allow": "Person", "for_value": parent_record.name}, "hide_descendants", 1
+		)
+		frappe.cache.delete_value("user_permissions")
+
+		visible_names_after_hide_descendants = frappe.get_list(
+			"Person",
+			fields=["person_name"],
+		)
+
+		self.assertEqual(visible_names[0], {"person_name": "Child"}, {"person_name": "Parent"})
```

Comment:
Assertion seems wrong, last param is supposed to be test failure method. ([ref](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual))